### DISCUSSION TITLE
two-way and also mesh communication

### DISCUSSION THREAD

**Discussion by hamidreza73** [pschatzmann/arduino-audio-tools#1038] [Q&A] [answered]
Thank you for the wonderful tool you provided us
I haven't delved into this library yet and haven't implemented any code, but before I start I want to ask a few questions.
My first question is, is it possible to establish a two-way communication through Bluetooth using this library and a ESP-a1s module? Similar to Bluetooth hands-free?! It means sink and source at the same time
And secondly, can ESP-a1s be connected to several devices with Bluetooth and the sound be taken from only one of them at the user's choice?

Let me explain both my questions by explaining my desired project:
I want to make a headphone that can be connected to my phone and laptop via Bluetooth, and the connection to both will be maintained at the same time. For example, I am watching a YouTube video on my laptop and I hear the sound from my headphones. Now my phone rings, and  just by pressing a button on the headphones, the sound of the video from the laptop is cut off, and the sound of the phone call is played through the headphones, and I can have a conversation.

Thank you for guiding me for this path, whether I can use this library or not? And what is the best approach to implement such system?
And finally, I must say that I may not know everything, but I want and can learn many things, and therefore I am eager to hear your tips and advice, from the simplest to the most professional.

thank you!

**Comment by pschatzmann** [comment, marked as answer]
The answer is still no! and the documentation of the A2DP library is stating __The Hands-Free Profile (HFP), Headset Profile (HSP) and stand alone AVRCP without A2DP are not supported!__

A more detailed answer you can find by looking at similar discussions or issues in the A2DP library...
